server/routes/updateGoal/updateGoalController.test.ts

```diff
@@ -157,6 +157,18 @@ describe('updateGoalController', () => {
         next as undefined as NextFunction,
       )
 
+      // Then
```

Comment:
I think this new `Then` block is in the wrong place (and might be masking a failing test - I'm not sure how this test passes as it currently stands ... 🤔 )

So, our Given When Then comment blocks are there to help us structure our tests and to aid readability etc. The basic idea is this:
* Given "a load of setup / preconditions / etc"
* When "I do one and only one thing (ie. the action under test)"
* Then "I make some assertions about what should have happened"

The idea is that we have 1 and only 1 of each comment block, and that they are in the correct sequence (G, W, T) 

So, your new assertion (expect that we've redirected to the review page) is the right thing to assert.
But we're asserting it before we've done the When .... which feels kinda weird. Other than being in the wrong place, I have idea how this passes, because surely jest should fail this expectation because we've not called `res.redirect` yet because we've not called the `submitReviewUpdateGoal` ..... oh, hang on, I've seen it!

Oh - you have 2 sets of When Then .... you have `Given, When, Then, When, Then` ! Ah ha!

We need to refactor this test because it's doing 2 things (2 `When` blocks)

Lets pair on this one later and I'll help you with it 👍

Reply:
Ah ha; I've sussed it - its actually 2 separate tests! That's often the clue - if there are 2 `Given`'s, it usually means its actually 2 tests. And that was the case here.
The first test was the one you were refactoring - the test of the `submitUpdateGoalForm` function - we simply needed to change that to not assert the service method was called to actually update the goal; and to change the page it redirects to. Because thats what the `submitUpdateGoalForm` does now 👍 

And then there is a new test for the new `submitReviewUpdateGoal` method. Bingo! That makes more sense 👍 

I've refactored your tests to reflect the above 👍